build.sbt

```diff
@@ -524,6 +524,29 @@ lazy val rspaceBench = (project in file("rspace-bench"))
   .enablePlugins(JmhPlugin)
   .dependsOn(rspace % "test->test", rholang % "test->test", models % "test->test")
 
+import scala.sys.process._
+lazy val secp256k1 =
+  (project in file("secp256k1"))
+    .settings(
+      commonSettings,
+      libraryDependencies ++= (commonDependencies :+ guava),
+      pullNativeLibs := {
+        if (java.nio.file.Files.notExists(new File(linux_x86_x64_local).toPath)) {
+          println("Missing linux_x86_64 native library, downloading...")
+          url(linux_x86_x64_remote) #> file(linux_x86_x64_local) !
+        }
+        if (java.nio.file.Files.notExists(new File(osx_x86_x64_local).toPath)) {
+          println("Missing osx_x86_64 native library, downloading...")
+          url(osx_x86_x64_remote) #> file(osx_x86_x64_local) !
+        }
+        if (java.nio.file.Files.notExists(new File(osx_aarch64_remote).toPath)) {
+          println("Missing osx_arm_64 native library, downloading...")
+          url(osx_aarch64_remote) #> file(osx_aarch64_local) !
+        }
+      },
+      Compile / compile := ((Compile / compile) dependsOn pullNativeLibs).value
+    )
```

Comment:
To be correct how to manage _generated_ files sbt provided official support.
https://www.scala-sbt.org/1.x/docs/Howto-Generating-Files.html#Generate+resources

Generated files should not reside in _src_ folder. We had similar problem with BNFC generated files before. Ypou can take a look how `FileFunction.cached` is used to generate files only once.
https://github.com/rchain/rchain/pull/3428/files#diff-ce9b359c2a460634e56ee6ed3774436704095e9eb1ba3be59fabac634e1b96faR84